client/setec/fields.go

```diff
@@ -0,0 +1,230 @@
+// Copyright (c) Tailscale Inc & AUTHORS
+// SPDX-License-Identifier: BSD-3-Clause
+
+package setec
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"path"
+	"reflect"
+	"strings"
+
+	"golang.org/x/exp/slices"
+)
+
+// ParseFields parses information about setec-tagged fields from v. The
+// concrete type of v must be a pointer to a struct value with at least one
+// tagged field.  The namePrefix, if non-empty, is joined to the front of each
+// tagged name, separated with a slash ("/").
+//
+// See [Fields] for a description of the struct tags and types recognized.
+func ParseFields(v any, namePrefix string) (*Fields, error) {
```

Comment:
I left it exported so that you can still apply secrets to a struct even if that struct isn't available when the store is constructed. I reworked the examples and the docs to focus on the pre-declared case, but in some of our existing use we wouldn't be able to do this, because we don't get the struct until later.